exercises/practice/react/lib/react.ex

```diff
@@ -0,0 +1,53 @@
+defmodule React do
+  defmodule InputCell do
+    defstruct [:name, :value]
+    @type t :: %InputCell{name: String.t(), value: any}
+  end
+
+  defmodule OutputCell do
+    defstruct [:name, :inputs, :compute, :value, :callbacks]
+
+    @type t :: %OutputCell{
+            name: String.t(),
+            inputs: [String.t()],
+            compute: (... -> any),
+            value: any,
+            callbacks: [String.t()]
+          }
+  end
+
+  @doc """
+  Start a reactive system
+  """
+  @spec new(cells :: [InputCell.t() | OutputCell.t()]) :: {:ok, pid}
+  def new(cells) do
+  end
+
+  @doc """
+  Return the value of an input or output cell
+  """
+  @spec get_value(cells :: pid, cell :: String.t()) :: any()
+  def get_value(cells, cell) do
+  end
+
+  @doc """
+  Set the value of an input cell
```

Comment:
Maybe this needs a hint about what the function returns?

Reply:
Ah, the callbacks. Yes, good idea.
Also I have a question about this. It is specified that you can only have specific callback once, so I picked a map as the data structure since the uniqueness of the keys would always enforce that. But that seems like a decision that has to be made by the student. Of course, it's also important for the tests so... I'm not sure.

Reply:
After some thought, I find it a bit weird that there are no actual functions involved in the callbacks 🤔. 
I'm thinking the API should rather be like this:

```
React.add_callback(cells, "always_two", fn value -> send(self(), {:callback1, value}))
:ok = React.set_value(cells, "input", 2)
refute_received {:callback, _}, "Expected callback not to be called" # maybe the failure message can be improved
```

Reply:
Ah nice! 
I actually don't have a good intuition of what callbacks are, so this is very helpful. This feels closer to what it should be!

Reply:
I had a go at something that I found sensible, please check :)

Reply:
Looks much better 👍